lib/nerves_time/system_time.ex

```diff
@@ -23,6 +23,12 @@ defmodule NervesTime.SystemTime do
     GenServer.start_link(__MODULE__, [], name: __MODULE__)
   end
 
+  def await_initialization(timeout) do
+    GenServer.call(__MODULE__, :await_async_init, timeout)
```

Comment:
I think this might need a little more thought. I see the idea here is that this call will only succeed with `NervesTime.SystemTime` has started - But I think this could still return earlier than you would like since all the work to initialize time is in a `handle_continue` callback which is to allow this GenServer to start responding while it is continuing init work.

Reply:
The GenServer won't respond to anything before the initial `handle_continue` execution is done. Returning with a continue value from `init/1` only unblocks whomever starts the process. Also`handle_continue` afaik was implemented exactly to prevent async setup (prev. often done with `Process.send_after`) from responding to things in the mailbox before setup has completed. `handle_continue` will always be executed directly after the callback it was initiated from without looking at the mailbox.

Reply:
Ah, perfect. TIL 🎉